### PR TITLE
C++: Repair support for createLSParser in the CWE-611 XXE query.

### PR DESCRIPTION
**Pull request by geoffw0**
Repair support for `createLSParser` in the CWE-611 XXE query, which it turned out was based on some incorrect information (see https://github.com/github/codeql/pull/9019).

Change note covered by the one in recent PR https://github.com/github/codeql/pull/8948.

### CODE REVIEW

File: cpp/ql/src/Security/CWE/CWE-611/XXE.ql [hunk 1]
```diff
@@ -217,12 +224,75 @@ class SetFeatureTranformer extends XXEFlowStateTranformer {
...
 
 /**
- * The `AbstractDOMParser.parse`, `SAXParser.parse` or `SAX2XMLReader.parse`
- * method.
+ * The `DOMLSParser.getDomConfig` function.
+ */
+class GetDomConfig extends Function {
+  GetDomConfig() {
+    this.hasName("getDomConfig") and
+    this.getDeclaringType() instanceof DomLSParserClass
```
Comment (MathiasVP) [inline cpp/ql/src/Security/CWE/CWE-611/XXE.ql:232]: We can use `getClassAndName` as well here, right?

Comment (geoffw0) [reply cpp/ql/src/Security/CWE/CWE-611/XXE.ql]: Yep, you have a good eye for spotting these.  Updated.

**Comment by geoffw0** [comment]
Looks the the DCA run has failed, so I'll report it to the DCA team and start another.

**Comment by geoffw0** [comment]
DCA LGTM.  The query is a fraction slower, but its far from concerning and to be expected when adding features.

**Pull request merged**